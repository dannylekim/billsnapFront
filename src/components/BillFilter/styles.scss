@import "../../variables";

div.bill__filter__container {
  top: 75px;
  height: 220px;
  width: 500px;
  border-radius: 6px;
  background-color: $light-gray-background;

  div.filter__header {
    text-align: left;
    margin-left: 10px;

    font-family: $primary-font;
    font-size: 25px;

    span.filter__header__title {
      margin-left: 10px;
      position: relative;
      top: 2px;
      font-weight: bold;
    }
  }

  div.filter__selections__container {
    margin-top: 10px;
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    height: 120px;
    width: 90%;
    margin-left: auto;
    margin-right: auto;

    span.badge.badge-outline-primary.badge-pill {
      border: 1px solid $secondary-text-color;
      box-sizing: border-box;
      border-radius: 50px;
      color: $secondary-text-color;
      font-family: $primary-font;
      font-style: normal;
      font-weight: normal;
      font-size: 20px;
      line-height: 16px;
      width: 120px;
      margin-bottom: 10px;

      &.active {
        background-color: $secondary-color;
      }
    }

    span.badge.badge-outline-primary.badge-pill:hover {
      background-color: #bebebe;
    }
  }

  div.advanced__filter__selection {
    position: relative;
    bottom: 80px;
    left: 380px;
    width: 180px;
    height: 110px;
    font-family: $primary-font;

    label.custom-control.custom-checkbox {
      padding: 0;
      font-style: normal;
      font-weight: normal;
      font-size: 15px;
      color: $secondary-text-color;
    }

    input.form-control {
      box-sizing: border-box;
      color: rgba(0, 0, 0, 0.42);
      font-size: 15px;
      background: $card-color;
    }

    div.date__filter {
      span.custom-control-description {
        margin-left: 25px;
      }

      input.form-control {
        width: 172px;
        border: 3px solid rgba(0, 0, 0, 0.17);
        background: $card-color;
      }
    }

    div.category__filter {
      input.form-control {
        margin-top: 10px;
        width: 150px;
        border: 1px solid #becad6;
        background: $card-color;
      }
    }
  }
}
