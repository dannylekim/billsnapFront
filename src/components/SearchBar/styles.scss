@import "../../variables";

#search__bill {
  display: grid;
  grid-template-rows: 1fr auto;
  grid-template-columns: 1fr auto;
}

#search__bill-options {
  grid-column-start: 2;
  grid-column-end: 3;
}

#search__bill-bar {
    grid-column-start: 1;
    grid-column-end: 3;
    border: 1px solid $light-gray-background;
    box-sizing: border-box;
    border-radius: 50px;
    background: $light-gray-background;
    display: flex;
    align-content: space-between;
}

  span.search__icon {
    width: 10%;
    border-top-left-radius: inherit;
    border-bottom-left-radius: inherit;
    border: inherit;
    background: inherit;
    margin-top: auto;
    margin-bottom: auto;
  }

  span.search__filter {
    width: 20%;
    margin-top: auto;
    margin-bottom: auto;
    font-weight: 500;
    font-size: 20px;
    border-top-right-radius: inherit;
    border-bottom-right-radius: inherit;

    span.simple__sort {
      margin-left: 10px;
    }
  }

  input.form-control {
    border: inherit;
    background: inherit;
    box-shadow: none;
  }