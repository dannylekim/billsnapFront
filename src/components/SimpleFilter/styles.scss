@import "../../variables";

div.quick__sorting__container {
    height: 120px;

    ul.sorting__content {
      width: 103px;
      border-radius: 6px;
      height: inherit;
      background-color: $light-gray-background;
      list-style: none;
      padding-top: 10px;
      padding-left: 0;

      li.sorting__titles {
        font-family: $primary-font;
        font-weight: 500;
        font-size: 20px;
        line-height: 26px;
        color: $secondary-text-color;
      }

      li.sorting__titles:hover {
        background-color: #bebebe;
      }
    }
  }