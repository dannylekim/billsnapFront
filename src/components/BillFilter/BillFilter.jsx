import React, { useRef } from "react";
import { Badge, FormCheckbox, FormInput } from "shards-react";

import { useOutsideAlerter } from "../../helpers/ClickEvent";
import "./styles.scss";

/**
 * @description the filter box when three horizontal lines clicked
 */
const BillFilter = ({
  activeTab,
  applyFiltering,
  currentSorting,
  billStatusFilter,
  closeHandler,
  dateCheckboxHandler,
  dateFilters,
  filterToggles,
  filterToggleChange,
  handleDateSelection,
  handleStatusChange,
}) => {
  const wrapperRef = useRef(null);
  useOutsideAlerter(wrapperRef, closeHandler);

  const filterBadges = [
    {
      id: "descending",
      title: "Newest",
      onClick: () => applyFiltering("Newest"),
    },
    {
      id: "ascending",
      title: "Oldest",
      onClick: () => applyFiltering("Oldest"),
    },
    {
      id: "by__status",
      title: "Status",
      onClick: () =>
        filterToggleChange({
          statusOpened: true,
          categoryOpened: false,
          dateOpened: false,
        }),
    },
    {
      id: "alphabetical__increasing",
      title: "A to Z",
      onClick: () => applyFiltering("A to Z"),
    },
    {
      id: "alphabetical__decreasing",
      title: "Z to A",
      onClick: () => applyFiltering("Z to A"),
    },
    {
      id: "by__category",
      title: "Category",
      onClick: () =>
        filterToggleChange({
          categoryOpened: true,
          statusOpened: false,
          dateOpened: false,
        }),
    },
    {
      id: "by__date",
      title: "Date",
      onClick: () =>
        filterToggleChange({
          dateOpened: true,
          categoryOpened: false,
          statusOpened: false,
        }),
    },
  ];

  const extraFilterSelected =
    filterToggles.dateOpened === true ||
    filterToggles.statusOpened === true ||
    filterToggles.categoryOpened === true;
  
  return (
    <div className='bill__filter__container' ref={wrapperRef}>
      <div className='filter__header'>
  <span className='filter__header__title'>Filters {activeTab}</span>
      </div>
      <div className='filter__selections__container'>
        {filterBadges.map((badgeObject, key) => (
          <Badge
            key={key}
            className={badgeObject.title === currentSorting && 'active'}
            outline
            pill
            id={badgeObject.id}
            onClick={badgeObject.onClick}>
            {badgeObject.title}
          </Badge>
        ))}
      </div>
      {extraFilterSelected && (
        <div className='advanced__filter__selection'>
          {filterToggles.dateOpened === true && (
            <div className='date__filter'>
              <FormCheckbox
                checked={dateFilters.startDate.selected}
                onChange={() => dateCheckboxHandler("startDate")}>
                Start date
                {dateFilters.startDate.value !== ""
                  ? `: ${dateFilters.startDate.value}`
                  : ""}
              </FormCheckbox>
              <FormCheckbox
                checked={dateFilters.endDate.selected}
                onChange={() => dateCheckboxHandler("endDate")}>
                End date
                {dateFilters.endDate.value !== ""
                  ? `: ${dateFilters.endDate.value}`
                  : ""}
              </FormCheckbox>
              <FormInput
                type='date'
                onChange={(event) => handleDateSelection(event)}
                disabled={
                  dateFilters.endDate.selected === false &&
                  dateFilters.startDate.selected === false
                }
              />
            </div>
          )}
          {filterToggles.statusOpened === true && (
            <div className='status__filter'>
              <FormCheckbox
                checked={billStatusFilter.resolved}
                onChange={() =>
                  handleStatusChange('resolved')
                }>
                PAID
              </FormCheckbox>
              <FormCheckbox
                checked={billStatusFilter.open}
                onChange={() =>
                  handleStatusChange('open')
                }>
                UNPAID
              </FormCheckbox>
              <FormCheckbox
                checked={billStatusFilter.in_progess}
                onChange={() =>
                  handleStatusChange('in_progess')
                }>
                ONGOING
              </FormCheckbox>
            </div>
          )}
          {filterToggles.categoryOpened === true && (
            <div className='category__filter'>
              <FormInput type='text' placeholder='Restaurant' />
            </div>
          )}
        </div>
      )}
    </div>
  );
};

export default BillFilter;
