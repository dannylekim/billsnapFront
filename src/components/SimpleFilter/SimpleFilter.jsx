import React, { useRef, useEffect } from "react";
import PropType from "prop-types";
import {useOutsideAlerter} from '../../helpers/ClickEvent'; 
import "./styles.scss";

export const SHORT_FILTER_OPTIONS = ["Newest", "A to Z", "Z to A", "Oldest"];

const SimpleFilter = ({
    applyFilter,
    currentActive,
    closeHandler
}) => {
  const wrapperRef = useRef(null);
  useOutsideAlerter(wrapperRef, closeHandler);

  return (
    <div className='quick__sorting__container' ref={wrapperRef}>
      <ul className='sorting__content'>
        { SHORT_FILTER_OPTIONS
          .filter((title) => title !== currentActive)
          .map((title, key) => (
            <li
              className='sorting__titles'
              onClick={() => applyFilter(title)}
              key={key}>
              {title}
            </li>
          ))}
      </ul>
    </div>
  );
};

SimpleFilter.propTypes = {
    applyFilter: PropType.func.isRequired,
    closeHandler: PropType.func.isRequired,
    currentActive: PropType.string
};

export default SimpleFilter;