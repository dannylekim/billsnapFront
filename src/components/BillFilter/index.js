import BillFilter from "./BillFilter.jsx";

export default BillFilter;
