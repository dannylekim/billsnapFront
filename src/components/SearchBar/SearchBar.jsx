import React, { Component } from "react";
import PropType from "prop-types";

import { FaBars, FaSearch } from "react-icons/fa";

import BillFilter from "../BillFilter";
import SimpleFilter from "../SimpleFilter";

import "./styles.scss";

class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentSorting: "Newest",
      filter: {
        type: "",
        statusOpened: false,
        categoryOpened: false,
        dateOpened: false,
      },
      dateFilters: {
        startDate: { selected: false, value: "" },
        endDate: { selected: false, value: "" },
      },
      billStatusFilter: { resolved: false, open: false, in_progess: false },

      toggle: {
        short: false,
        long: false,
      },
    };
  }

  /**
   * Close all opened filter/sorting popups
   */
  closeHandler = () => {
    this.setState({
      toggle: {
        short: false,
        long: false,
      },
    });
  };

  /**
   * Change date state on handler event
   */
  handleDateSelection = (event) => {
    event.preventDefault();
    const date = event.target.value;

    if (date !== "") {
      if (
        this.state.dateFilters.startDate.selected === true &&
        this.state.dateFilters.endDate.selected === false
      ) {
        this.setState({
          dateFilters: {
            ...this.state.dateFilters,
            startDate: { ...this.state.dateFilters.startDate, value: date },
          },
        });
      }

      if (
        this.state.dateFilters.endDate.selected === true &&
        this.state.dateFilters.startDate.selected === false
      ) {
        this.setState({
          dateFilters: {
            ...this.state.dateFilters,
            endDate: { ...this.state.dateFilters.endDate, value: date },
          },
        });
      }
    }
  };

  updateBills_TEST = (type) => {
    this.setState((prev) => ({
      sorting: { type, opened: false },
      filter: {
        ...prev.filter,
        opened: false,
        categoryOpened: false,
        statusOpened: false,
        dateOpened: false,
      },
      dateFilters: {
        ...prev.dateFilters,
        startDate: { ...prev.dateFilters.startDate, selected: false },
        endDate: { ...prev.dateFilters.endDate, selected: false },
      },
    }));

    if (type !== "A to Z" && type !== "Z to A") {
      const startDate = `start=${this.state.dateFilters.startDate.value}`;
      const endDate = `end=${this.state.dateFilters.endDate.value}`;

      const filterQueryParam = {
        Newest: `?${startDate}&${endDate}&sort_by=CREATED&order_by=DESC`,
        Oldest: `?${startDate}&${endDate}&sort_by=CREATED&order_by=ASC`,
      };
      this.props.fetchBills(filterQueryParam[type]);
    } else this.props.orderAlphabetical(type, this.props.bills);
  };

  updateBills(e) {
    console.log(e);
  }

  /**
   * Convert filtering data into REST params string
   */
  getFilterParams = () => {
    
  }

  /**
   * Convert sorting type into REST params string
   */
  static getSortingParams = (type) => {
    switch(type) {
      case "Newest":
        return `sort_by=CREATED&order_by=DESC`;
      case "Oldest":
        return `sort_by=CREATED&order_by=ASC`;
      case "A to Z":
        return 'sort_by=NAME&order_by=ASC';
      case "Z to A":
        return 'sort_by=NAME&order_by=DESC';
      default:
        return '';
    }
  }

  applySorting = (type) => {
    this.setState({
      currentSorting: type,
    });
    this.closeHandler();

    const params = this.constructor.getSortingParams(type);

    switch(this.props.activeTab) {
      case 'allBills':
        return this.props.fetchBills(`?${params}`);
      default:
        break;
    }

  }

  render() {
    const {
      onInputChangeHandler,
    } = this.props;

    const { dateFilters, filter, billStatusFilter, currentSorting } = this.state;

    return (
      <div id='search__bill'>
        <div id='search__bill-bar'>
          <span className='search__icon'>
            <FaSearch />
          </span>
          <input
            type='text'
            className='form-control border-0'
            onChange={onInputChangeHandler}
            placeholder='Search bill'
          />
          <span className='search__filter'>
            <span
              className='filter__bill'
              onClick={() =>
                this.setState((prev) => ({
                  toggle: { ...prev.toggle, long: true },
                }))
              }>
              <FaBars size={24} />
            </span>
            <span
              className='simple__sort'
              onClick={() =>
                this.setState((prev) => ({
                  toggle: { ...prev.toggle, short: true },
                }))
              }>
              {currentSorting}{" "}
            </span>
          </span>
        </div>
        <div id='search__bill-options'>
          {this.state.toggle.long && (
            <BillFilter
              dateFilters={dateFilters}
              filter={filter}
              billStatusFilter={billStatusFilter}
              handleDateSelection={this.handleDateSelection}
              updateBills={this.updateBills}
              setState={this.setState.bind(this)}
              closeHandler={this.closeHandler}
            />
          )}
          {this.state.toggle.short && (
            <SimpleFilter
              applyFilter={(e) => {
                /** TODO change bill filter */
                this.applySorting(e);
              }}
              closeHandler={this.closeHandler}
              currentActive={this.state.currentSorting}
            />
          )}
        </div>
      </div>
    );
  }
}

SearchBar.propTypes = {
  activeTab: PropType.string,
};

export default SearchBar;
