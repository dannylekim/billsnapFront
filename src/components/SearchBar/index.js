import SearchBar from './SearchBar';
import {connect} from "react-redux";

import {fetchMyBills, fetchPendingBills, updateSearchBillName} from "../../redux/actions/billActions";

const mapStateToProps = (state) => ({
    billNameSearch: state.bills.searchInput,
  });

const mapDispatchToProps = (dispatch) => ({
    fetchBills: (queryParam = "") => dispatch(fetchMyBills(queryParam)),
    fetchPendingBills: (queryParam = "?invitation_status=PENDING") => dispatch(fetchPendingBills(queryParam)),
    updateBillNameSearch: (input = "") => dispatch(updateSearchBillName(input)),
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);
