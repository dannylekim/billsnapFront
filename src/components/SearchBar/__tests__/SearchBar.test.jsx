import React from "react";
import SearchBar from "../SearchBar";
import BillFilter from "../../BillFilter";
import SimpleFilter from "../../SimpleFilter";

describe("SearchBar", () => {
  describe("render", () => {
    let handleMockFunction ,mockSortAlphabetical,mockSortingType, mockUpdateSearch,mockFetchBill;
    let wrapper, instance
    beforeEach(() => {
        handleMockFunction = jest.fn();
        mockUpdateSearch = jest.fn(e => e);
        mockFetchBill = jest.fn();
        mockSortingType = jest.fn();
        mockSortingType = "";

        wrapper = shallow(
          <SearchBar
            updateBillNameSearch={mockUpdateSearch}
            fetchBills={mockFetchBill}
            orderAlphabetical={mockSortAlphabetical}
            billNameSearch=""
            activeTab="allBills"
          />);
          instance = wrapper.instance();

    });

    describe("snapshots ðŸ“¸", () => {
      it("SearchBar should match snap shot without sorting type", () => {
        matches(wrapper);
      });
      it("SearchBar should match snap shot when sorting type is set", () => {
        wrapper.setState({currentSorting : "A to Z"})
        matches(wrapper);
      });
    });


    describe("components", () => {
      it("should change seachedQuery state on form search.", () => {
        const event = {
          target: { value: "hahahhahahahha" },
        };
        
        wrapper.find("input.form-control.border-0").simulate("change", event);
        expect(mockUpdateSearch).toBeCalledWith(event.target.value);
      });
  
      describe("short filter onClicks", () => {

        it("should set short filter as true", ()=> {
          wrapper.find("span.simple__sort").simulate("click");
          expect(wrapper.state().toggle.short).toBeTruthy();
        })

        // it("should trigger a Mock function when A to Z is clicked.", () => {
        //   expect(wrapper.find(SimpleFilter)).toHaveLength(1);
        // });

      });

      describe("long filter onClicks", () => {

        it("should set long filter as true", ()=> {
          wrapper.find("span.filter__bill").simulate("click");
          expect(wrapper.state().toggle.long).toBeTruthy();
        });

      });
  
      describe.skip("filter badge onClicks", () => {
        let billFilterWrapper, billWrapper, mockHandleDateSelection;
        beforeEach(() => {
           wrapper.setState((prev) => ({
             toggle: { ...prev.toggle, long: true },
           }))
           wrapper.update();
         
         
          mockHandleDateSelection = jest.fn();
          billFilterWrapper = () =>
            shallow(
              <BillFilter
                dateFilters={wrapper.state().dateFilters}
                filter={wrapper.state().filter}
                billStatusFilter={wrapper.state().billStatusFilter}
                handleDateSelection={mockHandleDateSelection}
                updateBills={(type) => instance.updateBills(type)}
                setState={(newState) => instance.setState(newState)}
              />
            );
          billWrapper = billFilterWrapper();
        });
  
        it("should close filter div", () => {
          expect(wrapper.state().toggle.long).toBeTruthy();
          billWrapper.find("div.filter__header").simulate("click");
          wrapper.update();
          expect(wrapper.state().toggle.long).toBeFalsy();
        });
  
        it("should trigger a Mock function when Newest is selected.", () => {
          billWrapper.find("Badge").at(0).simulate("click"); //calls updateBills
          expect(mockFetchBill).toHaveBeenCalled();
          expect(wrapper.state().currentSorting).toBe("Newest");
        });
  
        it("should trigger a Mock function when Oldest is clicked.", () => {
          billWrapper.find("Badge").at(1).simulate("click");
          expect(mockFetchBill).toHaveBeenCalled();
          expect(wrapper.state().currentSorting).toBe("Oldest");
        });
  
        it("should change status checkbox states on status filter select.", () => {
          expect(wrapper.state().filterToggles.statusOpened).toBeFalsy();
          billWrapper.find("Badge").at(2).simulate("click");
          expect(wrapper.state().filterToggles.statusOpened).toBeTruthy();
  
          billWrapper = billFilterWrapper();
  
          billWrapper.find("FormCheckbox").at(2).simulate("change");
          wrapper.update();
          expect(wrapper.state().statusFilter.in_progess).toBeTruthy();
          expect(wrapper.state().statusFilter.resolved).toBeFalsy();
          expect(wrapper.state().statusFilter.open).toBeFalsy();
  
          billWrapper.find("FormCheckbox").at(1).simulate("change");
          expect(wrapper.state().statusFilter.in_progess).toBeFalsy();
          expect(wrapper.state().statusFilter.resolved).toBeFalsy();
          expect(wrapper.state().statusFilter.open).toBeTruthy();
  
          billWrapper.find("FormCheckbox").at(0).simulate("change");
          expect(wrapper.state().statusFilter.in_progess).toBeFalsy();
          expect(wrapper.state().statusFilter.resolved).toBeTruthy();
          expect(wrapper.state().statusFilter.open).toBeFalsy();
        });
  
        it("should trigger a Mock function when A to Z is clicked.", () => {
          billWrapper.find("Badge").at(3).simulate("click");
          expect(mockSortAlphabetical).toHaveBeenCalled();
          expect(wrapper.state().sorting.type).toBe("A to Z");
        });
  
        it("should trigger a Mock function when Z to A is clicked.", () => {
          billWrapper.find("Badge").at(4).simulate("click");
          expect(mockSortAlphabetical).toHaveBeenCalled();
          expect(wrapper.state().currentSorting).toBe("Z to A");
        });
  
        it("should trigger category open state on category filter select", () => {
          expect(wrapper.state().filterToggles.categoryOpened).toBeFalsy();
          billWrapper.find("Badge").at(5).simulate("click");
          expect(wrapper.state().filterToggles.categoryOpened).toBeTruthy();
        });
  
        describe("testing the date function", () => {
          //has to re-shallow bill filter every time wrapper is updated (.update does not work on functional)
          beforeEach(() => {
            expect(wrapper.state().dateFilters.startDate.selected).toBeFalsy();
            expect(wrapper.state().dateFilters.endDate.selected).toBeFalsy();
  
            billWrapper.find("Badge").at(6).simulate("click");
  
            expect(wrapper.state().filterToggles.dateOpened).toBeTruthy();
          });
  
          describe(" Start date and End date should be triggered on handleDateSelection", () => {
            const event = {
              preventDefault: jest.fn(),
              target: {
                value: "2020-01-01",
              },
            };
  
            it("should change start date checkbox states onChange.", () => {
              billFilterWrapper().find("FormCheckbox").at(0).simulate("change");
              wrapper.update();
  
              billWrapper = billFilterWrapper();
              billWrapper.find("FormInput").simulate("change", event);
              const spy = jest.spyOn(instance, "handleDateSelection"); // handleDateSelection
              instance.handleDateSelection(event);
              wrapper.update();
              expect(wrapper.state().dateFilters.startDate.value).toBe(
                event.target.value
              );
  
              expect(spy).toBeCalledTimes(1);
              expect(spy).toHaveBeenCalledWith(event);
              spy.mockRestore();
            });
  
            it("should change enddate checkbox states onChange.", () => {
              billFilterWrapper().find("FormCheckbox").at(1).simulate("change");
              wrapper.update();
  
              billWrapper = billFilterWrapper();
  
              billWrapper.find("FormInput").simulate("change", event);
              instance.handleDateSelection(event);
              wrapper.update();
              expect(wrapper.state().dateFilters.endDate.value).toBe(
                event.target.value
              );
            });
          });
  
          it("should return null", () => {
            const event = {
              preventDefault: jest.fn(),
              target: {
                value: "",
              },
            };
            billFilterWrapper().find("FormCheckbox").at(1).simulate("change");
            wrapper.update();
  
            billFilterWrapper().find("FormInput").at(0).simulate("change", event);
            expect(instance.handleDateSelection(event)).toBe(null);
          });
        });
      });
    });
  });
});
