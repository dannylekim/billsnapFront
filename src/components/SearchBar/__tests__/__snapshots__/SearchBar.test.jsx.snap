// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar render snapshots ðŸ“¸ SearchBar should match snap shot when long filter is visible 1`] = `
<div
  id="search__bill"
>
  <div
    id="search__bill-bar"
  >
    <span
      className="search__icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </span>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search bill"
      type="text"
      value=""
    />
    <span
      className="search__filter"
    >
      <span
        className="filter__bill"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height={24}
          size={24}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </span>
      <span
        className="simple__sort"
        onClick={[Function]}
      >
        Newest
      </span>
    </span>
  </div>
  <div
    id="search__bill-options"
  >
    <div
      className="bill__filter__container"
    >
      <div
        className="filter__header"
      >
        <span
          className="filter__header__title"
        >
          Filters 
          allBills
        </span>
      </div>
      <div
        className="filter__selections__container"
      >
        <span
          className="active badge badge-outline-primary badge-pill"
          id="descending"
          onClick={[Function]}
        >
          Newest
        </span>
        <span
          className="badge badge-outline-primary badge-pill"
          id="ascending"
          onClick={[Function]}
        >
          Oldest
        </span>
        <span
          className="badge badge-outline-primary badge-pill"
          id="by__status"
          onClick={[Function]}
        >
          Status
        </span>
        <span
          className="badge badge-outline-primary badge-pill"
          id="alphabetical__increasing"
          onClick={[Function]}
        >
          A to Z
        </span>
        <span
          className="badge badge-outline-primary badge-pill"
          id="alphabetical__decreasing"
          onClick={[Function]}
        >
          Z to A
        </span>
        <span
          className="badge badge-outline-primary badge-pill"
          id="by__category"
          onClick={[Function]}
        >
          Category
        </span>
        <span
          className="badge badge-outline-primary badge-pill"
          id="by__date"
          onClick={[Function]}
        >
          Date
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchBar render snapshots ðŸ“¸ SearchBar should match snap shot when short filter is visible 1`] = `
<div
  id="search__bill"
>
  <div
    id="search__bill-bar"
  >
    <span
      className="search__icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </span>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search bill"
      type="text"
      value=""
    />
    <span
      className="search__filter"
    >
      <span
        className="filter__bill"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height={24}
          size={24}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </span>
      <span
        className="simple__sort"
        onClick={[Function]}
      >
        Newest
      </span>
    </span>
  </div>
  <div
    id="search__bill-options"
  >
    <div
      className="quick__sorting__container"
    >
      <ul
        className="sorting__content"
      >
        <li
          className="sorting__titles"
          onClick={[Function]}
        >
          A to Z
        </li>
        <li
          className="sorting__titles"
          onClick={[Function]}
        >
          Z to A
        </li>
        <li
          className="sorting__titles"
          onClick={[Function]}
        >
          Oldest
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`SearchBar render snapshots ðŸ“¸ SearchBar should match snap shot when sorting type is set 1`] = `
<div
  id="search__bill"
>
  <div
    id="search__bill-bar"
  >
    <span
      className="search__icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </span>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search bill"
      type="text"
      value=""
    />
    <span
      className="search__filter"
    >
      <span
        className="filter__bill"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height={24}
          size={24}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </span>
      <span
        className="simple__sort"
        onClick={[Function]}
      >
        A to Z
      </span>
    </span>
  </div>
  <div
    id="search__bill-options"
  />
</div>
`;

exports[`SearchBar render snapshots ðŸ“¸ SearchBar should match snap shot without sorting type 1`] = `
<div
  id="search__bill"
>
  <div
    id="search__bill-bar"
  >
    <span
      className="search__icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </span>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search bill"
      type="text"
      value=""
    />
    <span
      className="search__filter"
    >
      <span
        className="filter__bill"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height={24}
          size={24}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </span>
      <span
        className="simple__sort"
        onClick={[Function]}
      >
        Newest
      </span>
    </span>
  </div>
  <div
    id="search__bill-options"
  />
</div>
`;
